src/protobuf-net.Grpc.Reflection/ReflectionService.cs

```diff
@@ -172,7 +172,7 @@ public int Compare(FileDescriptorProto? left, FileDescriptorProto? right)
                     return 0;
                 }
 
-                return 1;
+                return -1;
```

Comment:
This was the bug :)

Reply:
I don't think this is a valid fix; consider two arbitrary descriptors without a dependency relation and with different names. With this as *either* 1 or -1, we get the scenario where `Compare(x,y)` would return -1 **and** `Compare(y,x)` would return -1, which is *not* a valid comparer.

IMO the lines 170-175 should be simply `return string.Compare(left.Name, right.Name)` ?

Reply:
```suggestion
                return string.Compare(left.Name, right.Name);
```

Reply:
String comparison is not needed in this case since this comparer is used to sort FileDescriptors in correct order. In gRPC reflection it is expected that FileDescriptors are received in correct order. Any dependencies must preceed a descriptor that has a depenency on it.

This fix is *should* work for transitive dependencies, as it will put Left on the "left" side of Right, thus Left will then be compared with the dependencies of Right. One could also consider a recursive dependency check in lines 162 and 166.

Reply:
in that case, isn't `return 0;` the more appropriate fix, which any stable sort implementation will interpret as "leave it in the original order"? note that LINQ `OrderBy` guarantees stable sort, but IIRC `List<T>` and `T[]` in-place sort *do not*

Reply:
Yes a `return 0;` and do a recursive dependency lookup here and here:
https://github.com/protobuf-net/protobuf-net.Grpc/blob/3ebd1a07c4520fdcc0fbf6271e12dbfc3f5160eb/src/protobuf-net.Grpc.Reflection/ReflectionService.cs#L162

https://github.com/protobuf-net/protobuf-net.Grpc/blob/3ebd1a07c4520fdcc0fbf6271e12dbfc3f5160eb/src/protobuf-net.Grpc.Reflection/ReflectionService.cs#L166

Should be a correct fix.

Reply:
Fixed.